pelican/readers.py

```diff
@@ -202,12 +202,20 @@ def __init__(self, *args, **kwargs):
 
     def __init__(self, *args, **kwargs):
         super(RstReader, self).__init__(*args, **kwargs)
+        self._source_path = None
 
     def _parse_metadata(self, document):
```

Comment:
Pass `source_path` to `_parse_metadata` instead. It doesn't make sense to store it in the instance. Reader instances aren't per file.

Reply:
Done.